tapir/shifts/templates/shifts/shift_calendar_template.html

```diff
@@ -10,56 +10,79 @@
 {% block content %}
-    <div class="card-group mb-2">
-        <div class="card">
-            <h5 class="card-header">
-                {% block card_title %}
-                {% endblock card_title %}
-            </h5>
-            <div class="card-body">
-                {% block card_helptext %}
-                {% endblock card_helptext %}
+    <div class="d-flex align-items-end flex-column">
+        <button class="btn btn-link"
+                data-bs-toggle="collapse"
+                data-bs-target="#collapseCalendarInfo"
+                aria-expanded="false"
+                style="text-decoration: none">
+            <div class="d-flex align-items-center" style="text-decoration: none">
+                <span style="font-size: 18px; text-decoration: none; color: black;">{% translate "Legend" %}</span>
+                <span class="material-icons" style="color: black">unfold_more</span>
             </div>
-        </div>
-        {% shift_filters %}
+        </button>
     </div>
-    <div class="card mb-2">
-        <h5 class="card-header">{% translate "Date range" %}</h5>
-        <div class="card-body d-flex gap-4">
-            <script>
-                function onFilter() {
-                    let url = "{% url "shifts:calendar" %}";
-                    url += "?date_from=" + document.getElementById("date-from").value;
-                    url += "&date_to=" + document.getElementById("date-to").value;
-                    window.location = url;
-                }
-            </script>
-            <div class="d-flex align-items-baseline gap-2">
-                <label for="date-from" class="form-label">{% translate "From" %}</label>
-                <input id="date-from"
-                       type="date"
-                       value="{{ date_from }}"
-                       class="form-control"
-                       style="max-width: 160px" />
-            </div>
-            <div class="d-flex align-items-baseline gap-2">
-                <label for="date-to" class="form-label">{% translate "To" %}</label>
-                <input id="date-to"
-                       type="date"
-                       value="{{ date_to }}"
-                       class="form-control"
-                       style="max-width: 160px" />
+    <div class="collapse" id="collapseCalendarInfo">
+        <div class="card-group mb-2">
+            <div class="card">
+                <h5 class="card-header">
+                    {% block card_title %}
+                    {% endblock card_title %}
+                </h5>
+                <div class="card-body">
+                    {% block card_helptext %}
+                    {% endblock card_helptext %}
+                </div>
             </div>
-            <button class="{% tapir_button_link %}" onclick="onFilter()">{% translate "Filter" %}</button>
+            {% shift_filters %}
         </div>
     </div>
     {% for week, shifts_by_day in shifts_by_weeks_and_days.items %}
```

Comment:
At least one `div` is not being closed inside this `for`. Your IDE should should an error about that. You can see that each week-card is inside the previous card if you scroll down a bit.